kernel/term_typing.ml

```diff
@@ -186,7 +192,8 @@ let rec unzip ctx j =
       unzip ctx { j with uj_val = mkApp (mkLambda (n,ty,j.uj_val),arg) }
 
 let hcons_j j =
-  { uj_val = hcons_constr j.uj_val; uj_type = hcons_constr j.uj_type} 
+  (** Do not hashcons type: it is not used by the callers of this function. *)
```

Comment:
How about renaming `hcons_j` to `hcons_term_in_judgement` or something similar? The name is a bit long but it reflects the non-standard behavior (hash consing only the term).